app/src/main/kotlin/com/wire/android/util/CurrentScreenManager.kt

```diff
@@ -45,18 +54,27 @@ class CurrentScreenManager @Inject constructor() : DefaultLifecycleObserver, Nav
 
     override fun onResume(owner: LifecycleOwner) {
         super.onResume(owner)
-        isAppVisibleFlow.value = true
+        appLogger.i("${TAG}: onResume called")
+        isAppVisibleFlow.value = true && isScreenUnLocked
         wasAppEverVisibleFlow.value = true
     }
 
     override fun onPause(owner: LifecycleOwner) {
         super.onPause(owner)
+        val pm: PowerManager? = context.getSystemService(Context.POWER_SERVICE) as PowerManager?
+        isScreenUnLocked = pm?.isInteractive ?: true
```

Comment:
`isScreenUnlocked` is only updated on `onPause`, really?

If it switches to `false` (screen locked).
On the next `onResume` it will be still false?

Will we have another `onPause` and `onResume` to refresh it then?

This feels weird.